src/measure_theory/function/ess_sup.lean

```diff
@@ -48,20 +48,56 @@ limsup_congr hfg
 lemma ess_inf_congr_ae {f g : α → β} (hfg : f =ᵐ[μ] g) :  ess_inf f μ = ess_inf g μ :=
 @ess_sup_congr_ae α βᵒᵈ _ _ _ _ _ hfg
 
+lemma ess_sup_const (c : β) (hμ : μ ≠ 0) : ess_sup (λ x : α, c) μ = c :=
+by { rw ←ae_ne_bot at hμ, exactI limsup_const c }
+
+lemma ess_inf_const (c : β) (hμ : μ ≠ 0) : ess_inf (λ x : α, c) μ = c :=
+by { rw ←ae_ne_bot at hμ, exactI liminf_const c }
+
 end conditionally_complete_lattice
 
 section conditionally_complete_linear_order
-variable [conditionally_complete_linear_order β]
+variables [conditionally_complete_linear_order β] {x : β} {f : α → β}
 
 lemma ess_sup_eq_Inf {m : measurable_space α} (μ : measure α) (f : α → β) :
   ess_sup f μ = Inf {a | μ {x | a < f x} = 0} :=
+by { dsimp [ess_sup, limsup, Limsup], simp only [ae_iff, not_le] }
+
+lemma ess_inf_eq_Sup {m : measurable_space α} (μ : measure α) (f : α → β) :
+  ess_inf f μ = Sup {a | μ {x | f x < a} = 0} :=
+by { dsimp [ess_inf, liminf, Liminf], simp only [ae_iff, not_le] }
+
+lemma ae_lt_of_ess_sup_lt (hx : ess_sup f μ < x)
+  (hf : is_bounded_under (≤) μ.ae f . is_bounded_default) : ∀ᵐ y ∂μ, f y < x :=
+filter.eventually_lt_of_limsup_lt hx hf
+
+lemma ae_lt_of_lt_ess_inf (hx : x < ess_inf f μ)
+  (hf : is_bounded_under (≥) μ.ae f . is_bounded_default) : ∀ᵐ y ∂μ, x < f y :=
+filter.eventually_lt_of_lt_liminf hx hf
+
+variables [topological_space β] [first_countable_topology β] [densely_ordered β] [order_topology β]
+
+lemma ae_le_of_ess_sup_le [no_max_order β] (hx : ess_sup f μ ≤ x)
```

Comment:
To expand, the lemma is also true for `ℕ` because for `n : ℕ` you can find a sequence `f` such that all `m > n` are less than or equal to some `f a`. But in some sense this is true for a different reason than for first countable dense orders. Also note that this isn't a general property of all orders since in the ordinal `ω₁ + 1` you have no sequence that's greater than all the things that are strictly less than `⊤` (cofinality troubles).

Reply:
I've pushed a proof. Either you have a next element after `a`, or you don't. In the latter case, you can use your proof. In the former, the proof is even easier. Feel free to clean it up, I haven't tried to make it pretty :-)
It's definitely not a property of all orders, it's where the first countable topology kicks in. But not the dense order.

I agree that dense order are the interesting case, but still if an assumption can be dispensed with it should be!

Reply:
Oh I see! I understood the proof on an order theoretic level but thought that the topology would mess up with my intuitive picture.